### PR TITLE
Change slugs to use underscore

### PR DESCRIPTION
**Pull request by itcyborg**
Update how handle is generated when creating an attribute. This is to prevent issues when parsing the attributes in javascript without adding additional mapping using php.
case : suppose we want to add an attribute named "warranty information", the handle generated will be 'warranty-information'. this commit attempts to fix this by replacing '-' with '_'

### CODE REVIEW

**Comment by itcyborg** [comment]
The snake helper does not change the delimiter if it already exists.
image

**Comment by itcyborg** [comment]
The suggestion can work if the words are separated by spaces.

**Comment by glennjacobs** [comment]
It's likely we may want to use this style of "slugification" in other parts of the system or even addons down the line. 

I suggest we look to add our own string helper to avoid duplication.

e.g. 

```php
      Str::macro('snakeSlug', function (string $value) {
            Str::slug($value, '_')
      });
```

**Comment by alecritson** [comment]
What if it was named something like:

```php
Str::handle($handle);
```

**Comment by itcyborg** [comment]
> What if it was named something like:
> 
> ```
> Str::handle($handle);
> ```

This would make more sense.

**Comment by glennjacobs** [comment]
Yeah, that name sounds good.

File: packages/admin/src/AdminHubServiceProvider.php [hunk 1]
```diff
@@ -131,6 +132,11 @@ public function boot()
             [SetStaffAuthMiddlewareListener::class, 'handle']
         );
 
+        // Handle generator
+        Str::macro('handle', function ($string) {
```
Comment (alecritson) [inline packages/admin/src/AdminHubServiceProvider.php:136]: I feel like this should be registered in the core. Other areas of GetCandy will likely benefit from this, not just admin hub code.

**Review by alecritson — CHANGES_REQUESTED** [review]
I appreciate your effort on this and it is looking good. 

Can we get some basic tests in for the macro to make sure it's returning what we expect based on a bunch of different strings that could be passed.

File: packages/admin/src/AdminHubServiceProvider.php (hunk 1 above)
Comment (itcyborg) [reply packages/admin/src/AdminHubServiceProvider.php]: I have updated as requested. The helper is now in getcandy core package. I've also added some unit test for the helper function. Hope this is well. let me know if further changes are required.

**Comment by alecritson** [comment]
Thanks for sorting @itcyborg I'll get this merged ready for the upcoming release!

**Review by alecritson — APPROVED** [review]

**Pull request merged**